jrqs-app/src/java/ua/dp/isd/jrqs/RequirementsService.java

```diff
@@ -27,22 +27,40 @@
 package ua.dp.isd.jrqs;
 
 import java.util.List;
+import ua.dp.isd.jrqs.ESearchBy.SearchBy;
 
 public class RequirementsService {
-    
-    public List<Requirement> getRequirementsByRowId(String RequirementRowIds) {
-        return new RequirementsDAO().getRequirementsByRowIds(RequirementRowIds);
-    }
-    
-    public List<Requirement> getRequirementsByRequirementNumbers(String RequirementNumbers, String LimitBySource) {
-        return new RequirementsDAO().getRequirementsByRequirementNumbers(RequirementNumbers, LimitBySource);
+
+    private final String rawParamsSearchBy;
+    private final String rawParamsUserInput;
+    private final String rawParamsLimitBySource;
+
+    public RequirementsService(String by, String value, String only) {
+        rawParamsSearchBy = by;
+        rawParamsUserInput = value;
+        rawParamsLimitBySource = only;
     }
-    
-    public List<Requirement> getRequirementsByTmsTaskNumbers(String TmsTaskNumbers, String LimitBySource) {
-        return new RequirementsDAO().getRequirementsByTmsTaskNumbers(TmsTaskNumbers, LimitBySource);
+
+    public List<Requirement> getRequirements() {
+        if (rawParamsSearchBy == null || rawParamsSearchBy.equals("")) {
+            throw new IllegalArgumentException("Invalid request: by parameter is missing in the request.");

```

Comment:
Yes, this is possible. The arguments are passed from the user. The thrown exception message will be displayed back to the user if he accidentally removed the required argument in the URL. In addition, do not you think that the method implementation should check all the method arguments before using them? I do not think it is good idea to check the arguments in the constructor of the class.